src/ir/item.rs

```diff
@@ -282,19 +282,7 @@ impl<'a> CanDeriveDefault<'a> for Item {
     type Extra = ();
 
     fn can_derive_default(&self, ctx: &BindgenContext, _: ()) -> bool {
-        ctx.options().derive_default &&
-        match self.kind {
-            ItemKind::Type(ref ty) => {
-                if self.is_opaque(ctx, &()) {
-                    ty.layout(ctx)
-                        .map_or(false,
-                                |l| l.opaque().can_derive_default(ctx, ()))
-                } else {
-                    ty.can_derive_default(ctx, self)
-                }
-            }
-            _ => false,
-        }
+        ctx.options().derive_default && ctx.lookup_item_id_can_derive_default(self.id())
```

Comment:
So there should only be two `CanDeriveDefault` impls left: `Item` and `ItemId` and neither use the `Extra` type parameter, so we should remove it now.